calour/filtering.py

```diff
@@ -430,51 +465,64 @@ def filter_samples(exp: Experiment, field, values, negate=False, inplace=False):
 
     '''
     values = _to_list(values)
-    return filter_by_metadata(exp, field=field, select=values,
-                              negate=negate, inplace=inplace)
+    return filter_by_metadata(exp, field=field, select=values, negate=negate, inplace=inplace)
 
 
 @ds.with_indent(4)
 @Experiment._record_sig
-def filter_abundance(exp: Experiment, min_abundance, **kwargs):
-    '''Filter keeping only features with >= min_abundance total over all samples
-    This is a convenience function wrapping filter_by_data()
+def filter_mean_abundance(exp: Experiment, cutoff=0.01, field=None, **kwargs):
+    '''Filter features with a mean at least cutoff of the mean total abundance/sample
+
+    For example, to keep features with mean abundance of 1% use `filter_abundance(cutoff=0.01)`.
 
     Parameters
     ----------
-    min_abundance : numeric
-        The minimal total abundance for each feature over all samples
+    cutoff : float, optional
+        The minimal mean abundance fraction (out of the mean of total
+        abundance per sample) for a feature in order to keep
+        it. Default is 0.01 - keep features with mean abundance >=1%
+        of mean total abundance per sample.
+    field : str or `None`, optional
+        The column in the sample_metadata. If it is not `None`, the
+        features that has mean abundance lower than the cutoff in *ALL*
+        sample categories in the specified sample_metadata column will
+        be filtered away.
 
     Keyword Arguments
     -----------------
     %(filtering.filter_by_data.parameters.negate)s
 
     Returns
-    -------
+    ------s-
     Experiment
-        the filtered object
 
     See Also
     --------
     filter_by_data
 
     '''
-    newexp = exp.filter_by_data('sum_abundance', axis=1, cutoff=min_abundance, **kwargs)
-    return newexp
+    if exp.normalized <= 0:
+        logger.warning('Do you forget to normalize your data? It is required before running this function')
+
+    cutoff = exp.normalized * cutoff
+
+    return exp.filter_by_data('mean_abundance', axis=1, field=field, cutoff=cutoff, **kwargs)
 
 
 @ds.with_indent(4)
 @Experiment._record_sig
-def filter_prevalence(exp: Experiment, fraction, cutoff=1/10000, **kwargs):
-    '''Filter features keeping only ones present in at least fraction fraction of the samples.
-    This is a convenience function wrapping filter_by_data()
+def filter_abundance(exp: Experiment, cutoff=10, **kwargs):
+    '''Filter features with a mean at least cutoff of the mean total abundance/sample
```

Comment:
I think this is a very confusing explanation (mean of the mean total abundance/sample). I couldn't understand it and i'm supposed to know what's going on :)
Maybe something like:
Filter features with sum over all samples at least cutoff.